File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Nodejs/core/generator/handler/Handlers.js
const SharedHandlerType = require("../internal/SharedHandlerType");
const CommandType = require("../../../utils/CommandType")
const SharedArgumentNameHandler = require("./shared/SharedArgumentNameHandler");
const SharedMethodNameHandler = require("./shared/SharedMethodNameHandler");
const SharedBodyHandler = require("./shared/body/SharedBodyHandler");
const SharedClassNameHandler = require("./shared/SharedClassNameHandler");
const SharedClassInstanceHandler = require("./shared/SharedClassInstanceHandler");
const SharedReturnTypeHandler = require("./shared/SharedReturnTypeHandler");
const SharedCreateInstanceBodyHandler = require("./shared/body/SharedCreateInstanceBodyHandler");
const SharedGetTypeBodyHandler = require("./shared/body/SharedGetTypeBodyHandler");
const SharedInvokeInstanceMethodBodyHandler = require("./shared/body/SharedInvokeInstanceMethodBodyHandler");
const SharedInvokeStaticMethodBodyHandler = require("./shared/body/SharedInvokeStaticMethodBodyHandler");
const SharedGetStaticFieldBodyHandler = require("./shared/body/SharedGetStaticFieldBodyHandler");
const SharedGetInstanceFieldBodyHandler = require("./shared/body/SharedGetInstanceFieldBodyHandler");
const CreateInstanceGeneratorHandler = require("./CreateInstanceGeneratorHandler");
const GetTypeGeneratorHandler = require("./GetTypeGeneratorHandler");
const InvokeInstanceMethodGeneratorHandler = require("./InvokeInstanceMethodGeneratorHandler");
const InvokeStaticMethodGeneratorHandler = require("./InvokeStaticMethodGeneratorHandler");
const GetStaticFieldGeneratorHandler = require("./GetStaticFieldGeneratorHandler");
const GetInstanceFieldGeneratorHandler = require("./GetInstanceFieldGeneratorHandler");

class Handlers {
    GENERATOR_HANDLER = {};
    SHARED_HANDLER = {};
    SHARED_BODY_HANDLER = {};

    constructor() {
        this.SHARED_HANDLER[SharedHandlerType.ARGUMENT_NAME] = new SharedArgumentNameHandler()
        this.SHARED_HANDLER[SharedHandlerType.METHOD_BODY] = new SharedBodyHandler()
        this.SHARED_HANDLER[SharedHandlerType.CLASS_NAME] = new SharedClassNameHandler()
        this.SHARED_HANDLER[SharedHandlerType.METHOD_NAME] = new SharedMethodNameHandler()
        this.SHARED_HANDLER[SharedHandlerType.CLAZZ_INSTANCE] = new SharedClassInstanceHandler()
        this.SHARED_HANDLER[SharedHandlerType.RETURN_TYPE] = new SharedReturnTypeHandler()

        this.SHARED_BODY_HANDLER[CommandType.CreateClassInstance] = new SharedCreateInstanceBodyHandler()
        this.SHARED_BODY_HANDLER[CommandType.GetType] = new SharedGetTypeBodyHandler()
        this.SHARED_BODY_HANDLER[CommandType.InvokeInstanceMethod] = new SharedInvokeInstanceMethodBodyHandler()
        this.SHARED_BODY_HANDLER[CommandType.InvokeStaticMethod] = new SharedInvokeStaticMethodBodyHandler()
        this.SHARED_BODY_HANDLER[CommandType.GetStaticField] = new SharedGetStaticFieldBodyHandler()
        this.SHARED_BODY_HANDLER[CommandType.GetInstanceField] = new SharedGetInstanceFieldBodyHandler()

        this.GENERATOR_HANDLER[CommandType.CreateClassInstance] = new CreateInstanceGeneratorHandler()
        this.GENERATOR_HANDLER[CommandType.GetType] = new GetTypeGeneratorHandler()
        this.GENERATOR_HANDLER[CommandType.InvokeInstanceMethod] = new InvokeInstanceMethodGeneratorHandler()
        this.GENERATOR_HANDLER[CommandType.InvokeStaticMethod] = new InvokeStaticMethodGeneratorHandler()
        this.GENERATOR_HANDLER[CommandType.GetStaticField] = new GetStaticFieldGeneratorHandler()
        this.GENERATOR_HANDLER[CommandType.GetInstanceField] = new GetInstanceFieldGeneratorHandler()
    }
}

module.exports = Handlers

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Python/javonet/core/generator/handler/Handlers.py
from javonet.core.generator.handler.CreateInstanceGeneratorHandler import CreateInstanceGeneratorHandler
from javonet.core.generator.handler.GetInstanceFieldGeneratorHandler import GetInstanceFieldGeneratorHandler
from javonet.core.generator.handler.GetStaticFieldGeneratorHandler import GetStaticFieldGeneratorHandler
from javonet.core.generator.handler.GetTypeGeneratorHandler import GetTypeGeneratorHandler
from javonet.core.generator.handler.InvokeInstanceMethodGeneratorHandler import InvokeInstanceMethodGeneratorHandler
from javonet.core.generator.handler.InvokeStaticMethodGeneratorHandler import InvokeStaticMethodGeneratorHandler
from javonet.core.generator.handler.shared.body.SharedBodyHandler import SharedBodyHandler
from javonet.core.generator.handler.shared.body.SharedCreateInstanceBodyHandler import SharedCreateInstanceBodyHandler
from javonet.core.generator.handler.shared.body.SharedGetInstanceFieldBodyHandler import \
    SharedGetInstanceFieldBodyHandler
from javonet.core.generator.handler.shared.body.SharedGetStaticFieldBodyHandler import SharedGetStaticFieldBodyHandler
from javonet.core.generator.handler.shared.body.SharedGetTypeBodyHandler import SharedGetTypeBodyHandler
from javonet.core.generator.handler.shared.body.SharedInvokeInstanceMethodBodyHandler import \
    SharedInvokeInstanceMethodBodyHandler
from javonet.core.generator.handler.shared.body.SharedInvokeStaticMethodBodyHandler import \
    SharedInvokeStaticMethodBodyHandler
from javonet.utils.CommandType import CommandType

from javonet.core.generator.handler.shared.SharedArgumentNameHandler import SharedArgumentNameHandler
from javonet.core.generator.handler.shared.SharedClassInstanceHandler import SharedClassInstanceHandler
from javonet.core.generator.handler.shared.SharedClassNameHandler import SharedClassNameHandler
from javonet.core.generator.handler.shared.SharedMethodNameHandler import SharedMethodNameHandler
from javonet.core.generator.handler.shared.SharedModifierHandler import SharedModifierHandler
from javonet.core.generator.handler.shared.SharedReturnTypeHandler import SharedReturnTypeHandler
from javonet.core.generator.handler.shared.SharedTypeHandler import SharedTypeHandler
from javonet.core.generator.internal.SharedHandlerType import SharedHandlerType


class Handlers:

    types_to_analyze = []

    GENERATOR_HANDLER = dict()
    SHARED_HANDLER = dict()
    SHARED_BODY_HANDLER = dict()

    def __init__(self):
        self.SHARED_HANDLER[SharedHandlerType.ARGUMENT_NAME] = SharedArgumentNameHandler()
        self.SHARED_HANDLER[SharedHandlerType.METHOD_BODY] = SharedBodyHandler()
        self.SHARED_HANDLER[SharedHandlerType.CLASS_NAME] = SharedClassNameHandler()
        self.SHARED_HANDLER[SharedHandlerType.METHOD_NAME] = SharedMethodNameHandler()
        self.SHARED_HANDLER[SharedHandlerType.MODIFIER] = SharedModifierHandler()
        self.SHARED_HANDLER[SharedHandlerType.TYPE] = SharedTypeHandler()
        self.SHARED_HANDLER[SharedHandlerType.CLAZZ_INSTANCE] = SharedClassInstanceHandler()
        self.SHARED_HANDLER[SharedHandlerType.RETURN_TYPE] = SharedReturnTypeHandler()

        self.SHARED_BODY_HANDLER[CommandType.CreateClassInstance] = SharedCreateInstanceBodyHandler()
        self.SHARED_BODY_HANDLER[CommandType.GetType] = SharedGetTypeBodyHandler()
        self.SHARED_BODY_HANDLER[CommandType.InvokeInstanceMethod] = SharedInvokeInstanceMethodBodyHandler()
        self.SHARED_BODY_HANDLER[CommandType.InvokeStaticMethod] = SharedInvokeStaticMethodBodyHandler()
        self.SHARED_BODY_HANDLER[CommandType.GetStaticField] = SharedGetStaticFieldBodyHandler()
        self.SHARED_BODY_HANDLER[CommandType.GetInstanceField] = SharedGetInstanceFieldBodyHandler()

        self.GENERATOR_HANDLER[CommandType.CreateClassInstance] = CreateInstanceGeneratorHandler()
        self.GENERATOR_HANDLER[CommandType.GetType] = GetTypeGeneratorHandler()
        self.GENERATOR_HANDLER[CommandType.InvokeInstanceMethod] = InvokeInstanceMethodGeneratorHandler()
        self.GENERATOR_HANDLER[CommandType.InvokeStaticMethod] = InvokeStaticMethodGeneratorHandler()
        self.GENERATOR_HANDLER[CommandType.GetStaticField] = GetStaticFieldGeneratorHandler()
        self.GENERATOR_HANDLER[CommandType.GetInstanceField] = GetInstanceFieldGeneratorHandler()
